internal/searcher/model.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2023 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+
+package searcher
+
+type NixPackageInfoList []*NixPackageInfo
+
+type NixPackageInfo struct {
+	AttributePath string  `json:"attribute_path,omitempty"`
+	NixpkgCommit  string  `json:"nixpkg_commit,omitempty"`
+	PName         string  `json:"pname,omitempty"`
+	Version       string  `json:"version,omitempty"`
+	Score         float64 `json:"score,omitempty"`
+}
+
+type Result struct {
+	Name     string             `json:"name"`
+	Packages NixPackageInfoList `json:"packages"`
```

Comment:
just curious on why not use a `[]*PackageInfo` here, instead of introducing a new type?

Reply:
This is copied over from the backend  (the backend will import this struct once it is merged). I added it mostly for readability, since in some places we have maps of lists and lists of lists so it can get quite confusing.